### ISSUE TITLE
Ticktock: ability to add other than timedelta

### ISSUE THREAD

**Issue by jtniehof** [spacepy/spacepy#317]
Ticktock add/subtract methods operate on timedeltas and thus fall through to Python datetimes, and can't properly handle leapseconds. Should add ability to use other types (either seconds or Julian days as floats?).

**Comment by balarsen** [comment]
Can the numpy datetime64 do leap seconds? Would really love to find another
implementation that we can use or copy.

**Comment by jtniehof** [comment]
Nope, it can't. I've argued with them about it but they've said they won't support it unless I can tell them what (2050, 1, 1) - (2049, 12, 31, 23, 59, 59) is, right now. They don't get that UTC isn't defined in the future.

**Comment by balarsen** [comment]
That is frustrating. We need to think hard on how we handle it. For differences a conversion to TAI and add/subtract is great and should work. Our implementation of checking for leap seconds in the UTC->TAI conversion is a huge bottleneck currently. 
https://github.com/spacepy/spacepy/blob/0fbb879a5d9f50fa2d22bf5b056ba60f1edac7b6/spacepy/time.py#L1281

**Comment by jtniehof** [comment]
My first thought would be to allow adding numerical only for numerical representations (TAI, MJD, GPS, CDF Epoch) and have it operate on the "native" representation of the Ticktock. So if you add 1 to MJD it's a day, 1 to TAI is a second, 1 to Epoch is a millisecond. Alternatively I'm okay with saying it's always seconds.

**Comment by balarsen** [comment]
My take would be to decide on an internal time base (I vote TAI) and just always convert to that and do all calculations there then transform to the output. From there math is seconds or with timedelta objects would be pretty easy. 

This implies we need the input to TAI conversion to be as efficient as possible.
